### PR TITLE
Add Flag to control Dropout() behavior at test time

### PR DESCRIPTION
**Pull request by Pyrestone**
### Summary
Adds a Flag, called `drop_at_test_time` to the Dropout layer.
This flag allows users to explicitly to apply the training phase dropout behavior at test time.
The motivation behind this mainly comes from [this paper](https://arxiv.org/abs/1506.02142), which uses Dropout at test time to obtain uncertainty estimates for neuralnetwork predictions.
The default behavior of the Dropout layer remains unchanged, since the argument is set to `False` by default.
### Related Issues
In #5357, the desire for more control over dropout behavior at test time has been voiced. Currently, the recommended solution is hacky and requires the use of undocumented features of the Keras API, either by creating custom layers or by 
### PR Overview

- [**Y**] This PR requires new unit tests [y/n] -> I am not sure how to create appropriate tests, suggestions are welcome.
- [Y] This PR requires to update the documentation [y/n] -> Documentation has been updated accordingly.
- [**N**] This PR is backwards compatible [y/n] -> Default behavior is backwards compatible, newly introduced behavior adds the flag to `get_config()` and thus prevents loading the layer in previous versions of Keras.
- [**Y**] This PR changes the current API [y/n] (all API changes need to be approved by fchollet) -> non-breaking changes to the API are introduced via a new optional argument.

### CODE REVIEW

**Comment by fchollet** [comment]
You can do:

```python
y = Dropout(0.5)(x, training=True)
```

to get dropout at both training and test time.

**Comment by Pyrestone** [comment]
> 
> 
> You can do:
> 
> ```python
> y = Dropout(0.5)(x, training=True)
> ```
> 
> to get dropout at both training and test time.

Does this persist through saving and loading the model?
